libr/anal/fcn.c

```diff
@@ -353,7 +353,7 @@ static ut64 search_reg_val(RAnal *anal, ut8 *buf, ut64 len, ut64 addr, char *reg
 		0
 	};
 	ut64 ret = UT64_MAX;
-	for (offs = 0; offs < len; offs += anal->addrbytes * oplen) {
+	for (offs = 0; offs < len; offs += anal->iob.io->addrbytes * oplen) {
```

Comment:
asuming io is not null and accessing all those pointers of pointers on each iteration is unnecessary hold that value before entering the loop in a const int which defaults to 1 if io is null

Reply:
Can you elaborate?